src/transaction/transaction.ts

```diff
@@ -24,7 +24,7 @@ const TxByronWitness = (
 const TxShelleyWitness = (publicKey: Buffer, signature: Buffer): TxWitnessShelley => [publicKey, signature]
 
 const TxAux = (unsignedTxCborHex: UnsignedTxCborHex): _TxAux => {
-  const unsignedTxDecoded = decodeCbor(unsignedTxCborHex)
+  const unsignedTxDecoded = deconstructUnsignedTxDecoded(decodeCbor(unsignedTxCborHex))
```

Comment:
I would say that the checks from `deconstructUnsignedTxDecoded` belong to `isUnsignedTxDecoded` (which can throw if the tx contains elements that are not supported), or we can have two such functions, but with similar API: both would accept cbor-decoded unsignedTxDecoded and the first one will check validity and the other whether the features are supported (although I don't see why this would be better than a single validation function, the unsupported features check is just some kind of enhancement, like saying to the user "wrong input because of this" and not just "wrong input").

Reply:
`isUnsignedTxDecoded` is guard and we use guards only to validate if the expected type is really the type we expect it to be. Meanwhile logic inside `deconstructUnsignedTxDecoded` actually transforms the array and possibly throws an error. It doesn't seem right to me to move it there.